<!DOCTYPE html><html lang="zh-CN" data-theme="light"><head><meta charset="UTF-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0,viewport-fit=cover"><title>Python-StudyNotes-GWASlab | FarMountBiolog</title><meta name="author" content="远山"><meta name="copyright" content="远山"><meta name="format-detection" content="telephone=no"><meta name="theme-color" content="#ffffff"><meta name="description" content="Thesis：GWASLab preprint: He, Y., Koido, M., Shimmori, Y., Kamatani, Y. (2023). GWASLab: a Python package for processing and visualizing GWAS summary statistics. Preprint at Jxiv, 2023-5. https:&#x2F;&#x2F;doi.o">
<meta property="og:type" content="article">
<meta property="og:title" content="Python-StudyNotes-GWASlab">
<meta property="og:url" content="http://example.com/2025/03/19/Python-StudyNotes-GWASlab/index.html">
<meta property="og:site_name" content="FarMountBiolog">
<meta property="og:description" content="Thesis：GWASLab preprint: He, Y., Koido, M., Shimmori, Y., Kamatani, Y. (2023). GWASLab: a Python package for processing and visualizing GWAS summary statistics. Preprint at Jxiv, 2023-5. https:&#x2F;&#x2F;doi.o">
<meta property="og:locale" content="zh_CN">
<meta property="og:image" content="http://example.com/img/Kakasi.jpg">
<meta property="article:published_time" content="2025-03-19T14:00:24.000Z">
<meta property="article:modified_time" content="2025-03-28T02:13:05.339Z">
<meta property="article:author" content="远山">
<meta property="article:tag" content="Pythony">
<meta name="twitter:card" content="summary">
<meta name="twitter:image" content="http://example.com/img/Kakasi.jpg"><link rel="shortcut icon" href="/img/Celtics.jpg"><link rel="canonical" href="http://example.com/2025/03/19/Python-StudyNotes-GWASlab/index.html"><link rel="preconnect" href="//cdn.jsdelivr.net"/><link rel="preconnect" href="//busuanzi.ibruce.info"/><link rel="stylesheet" href="/css/index.css?v=4.13.0"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fortawesome/fontawesome-free@6.5.1/css/all.min.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fancyapps/ui@5.0.33/dist/fancybox/fancybox.min.css" media="print" onload="this.media='all'"><script>const GLOBAL_CONFIG = {
  root: '/',
  algolia: undefined,
  localSearch: {"path":"/search.xml","preload":false,"top_n_per_article":1,"unescape":false,"languages":{"hits_empty":"找不到您查询的内容：${query}","hits_stats":"共找到 ${hits} 篇文章"}},
  translate: undefined,
  noticeOutdate: undefined,
  highlight: {"plugin":"highlight.js","highlightCopy":true,"highlightLang":false,"highlightHeightLimit":false},
  copy: {
    success: '复制成功',
    error: '复制错误',
    noSupport: '浏览器不支持'
  },
  relativeDate: {
    homepage: false,
    post: false
  },
  runtime: '',
  dateSuffix: {
    just: '刚刚',
    min: '分钟前',
    hour: '小时前',
    day: '天前',
    month: '个月前'
  },
  copyright: undefined,
  lightbox: 'mediumZoom',
  Snackbar: undefined,
  infinitegrid: {
    js: 'https://cdn.jsdelivr.net/npm/@egjs/infinitegrid@4.11.1/dist/infinitegrid.min.js',
    buttonText: '加载更多'
  },
  isPhotoFigcaption: true,
  islazyload: false,
  isAnchor: false,
  percent: {
    toc: true,
    rightside: false,
  },
  autoDarkmode: false
}</script><script id="config-diff">var GLOBAL_CONFIG_SITE = {
  title: 'Python-StudyNotes-GWASlab',
  isPost: true,
  isHome: false,
  isHighlightShrink: false,
  isToc: true,
  postUpdate: '2025-03-28 10:13:05'
}</script><script>(win=>{
      win.saveToLocal = {
        set: (key, value, ttl) => {
          if (ttl === 0) return
          const now = Date.now()
          const expiry = now + ttl * 86400000
          const item = {
            value,
            expiry
          }
          localStorage.setItem(key, JSON.stringify(item))
        },
      
        get: key => {
          const itemStr = localStorage.getItem(key)
      
          if (!itemStr) {
            return undefined
          }
          const item = JSON.parse(itemStr)
          const now = Date.now()
      
          if (now > item.expiry) {
            localStorage.removeItem(key)
            return undefined
          }
          return item.value
        }
      }
    
      win.getScript = (url, attr = {}) => new Promise((resolve, reject) => {
        const script = document.createElement('script')
        script.src = url
        script.async = true
        script.onerror = reject
        script.onload = script.onreadystatechange = function() {
          const loadState = this.readyState
          if (loadState && loadState !== 'loaded' && loadState !== 'complete') return
          script.onload = script.onreadystatechange = null
          resolve()
        }

        Object.keys(attr).forEach(key => {
          script.setAttribute(key, attr[key])
        })

        document.head.appendChild(script)
      })
    
      win.getCSS = (url, id = false) => new Promise((resolve, reject) => {
        const link = document.createElement('link')
        link.rel = 'stylesheet'
        link.href = url
        if (id) link.id = id
        link.onerror = reject
        link.onload = link.onreadystatechange = function() {
          const loadState = this.readyState
          if (loadState && loadState !== 'loaded' && loadState !== 'complete') return
          link.onload = link.onreadystatechange = null
          resolve()
        }
        document.head.appendChild(link)
      })
    
      win.activateDarkMode = () => {
        document.documentElement.setAttribute('data-theme', 'dark')
        if (document.querySelector('meta[name="theme-color"]') !== null) {
          document.querySelector('meta[name="theme-color"]').setAttribute('content', '#0d0d0d')
        }
      }
      win.activateLightMode = () => {
        document.documentElement.setAttribute('data-theme', 'light')
        if (document.querySelector('meta[name="theme-color"]') !== null) {
          document.querySelector('meta[name="theme-color"]').setAttribute('content', '#ffffff')
        }
      }
      const t = saveToLocal.get('theme')
    
        if (t === 'dark') activateDarkMode()
        else if (t === 'light') activateLightMode()
      
      const asideStatus = saveToLocal.get('aside-status')
      if (asideStatus !== undefined) {
        if (asideStatus === 'hide') {
          document.documentElement.classList.add('hide-aside')
        } else {
          document.documentElement.classList.remove('hide-aside')
        }
      }
    
      const detectApple = () => {
        if(/iPad|iPhone|iPod|Macintosh/.test(navigator.userAgent)){
          document.documentElement.classList.add('apple')
        }
      }
      detectApple()
    })(window)</script><meta name="generator" content="Hexo 7.3.0"></head><body><div id="sidebar"><div id="menu-mask"></div><div id="sidebar-menus"><div class="avatar-img is-center"><img src="/img/loading.gif" data-original="/img/Kakasi.jpg" onerror="onerror=null;src='/img/friend_404.gif'" alt="avatar"/></div><div class="sidebar-site-data site-data is-center"><a href="/archives/"><div class="headline">文章</div><div class="length-num">10</div></a><a href="/tags/"><div class="headline">标签</div><div class="length-num">4</div></a><a href="/categories/"><div class="headline">分类</div><div class="length-num">0</div></a></div><hr class="custom-hr"/><div class="menus_items"><div class="menus_item"><a class="site-page" href="/"><i class="fa-fw fas fa-home"></i><span> 主页</span></a></div><div class="menus_item"><a class="site-page group" href="javascript:void(0);"><i class="fa-fw fa fa-graduation-cap"></i><span> 博文</span><i class="fas fa-chevron-down"></i></a><ul class="menus_item_child"><li><a class="site-page child" href="/archives/"><i class="fa-fw fa fa-folder-open"></i><span> 归档</span></a></li></ul></div><div class="menus_item"><a class="site-page" href="/comment/"><i class="fa-fw fa fa-paper-plane"></i><span> 留言板</span></a></div><div class="menus_item"><a class="site-page" href="/about/"><i class="fa-fw fas fa-heart"></i><span> 关于笔者</span></a></div></div></div></div><div class="post" id="body-wrap"><header class="post-bg" id="page-header" style="background-image: url('/img/pic_1.png')"><nav id="nav"><span id="blog-info"><a href="/" title="FarMountBiolog"><span class="site-name">FarMountBiolog</span></a></span><div id="menus"><div id="search-button"><a class="site-page social-icon search" href="javascript:void(0);"><i class="fas fa-search fa-fw"></i><span> 搜索</span></a></div><div class="menus_items"><div class="menus_item"><a class="site-page" href="/"><i class="fa-fw fas fa-home"></i><span> 主页</span></a></div><div class="menus_item"><a class="site-page group" href="javascript:void(0);"><i class="fa-fw fa fa-graduation-cap"></i><span> 博文</span><i class="fas fa-chevron-down"></i></a><ul class="menus_item_child"><li><a class="site-page child" href="/archives/"><i class="fa-fw fa fa-folder-open"></i><span> 归档</span></a></li></ul></div><div class="menus_item"><a class="site-page" href="/comment/"><i class="fa-fw fa fa-paper-plane"></i><span> 留言板</span></a></div><div class="menus_item"><a class="site-page" href="/about/"><i class="fa-fw fas fa-heart"></i><span> 关于笔者</span></a></div></div><div id="toggle-menu"><a class="site-page" href="javascript:void(0);"><i class="fas fa-bars fa-fw"></i></a></div></div></nav><div id="post-info"><h1 class="post-title">Python-StudyNotes-GWASlab</h1><div id="post-meta"><div class="meta-firstline"><span class="post-meta-date"><i class="far fa-calendar-alt fa-fw post-meta-icon"></i><span class="post-meta-label">发表于</span><time class="post-meta-date-created" datetime="2025-03-19T14:00:24.000Z" title="发表于 2025-03-19 22:00:24">2025-03-19</time><span class="post-meta-separator">|</span><i class="fas fa-history fa-fw post-meta-icon"></i><span class="post-meta-label">更新于</span><time class="post-meta-date-updated" datetime="2025-03-28T02:13:05.339Z" title="更新于 2025-03-28 10:13:05">2025-03-28</time></span></div><div class="meta-secondline"><span class="post-meta-separator">|</span><span class="post-meta-wordcount"><i class="far fa-file-word fa-fw post-meta-icon"></i><span class="post-meta-label">字数总计:</span><span class="word-count">1.1k</span><span class="post-meta-separator">|</span><i class="far fa-clock fa-fw post-meta-icon"></i><span class="post-meta-label">阅读时长:</span><span>5分钟</span></span><span class="post-meta-separator">|</span><span class="post-meta-pv-cv" id="" data-flag-title="Python-StudyNotes-GWASlab"><i class="far fa-eye fa-fw post-meta-icon"></i><span class="post-meta-label">阅读量:</span><span id="busuanzi_value_page_pv"><i class="fa-solid fa-spinner fa-spin"></i></span></span></div></div></div></header><main class="layout" id="content-inner"><div id="post"><article class="post-content" id="article-container"><p>Thesis：GWASLab preprint: He, Y., Koido, M., Shimmori, Y., Kamatani, Y. (2023). GWASLab: a Python package for processing and visualizing GWAS summary statistics. Preprint at Jxiv, 2023-5. <a target="_blank" rel="noopener" href="https://doi.org/10.51094/jxiv.370">https://doi.org/10.51094/jxiv.370</a></p>
<h1 id="1-GWAS可视化学习"><a href="#1-GWAS可视化学习" class="headerlink" title="1. GWAS可视化学习"></a>1. GWAS可视化学习</h1><p>原文链接：<a target="_blank" rel="noopener" href="https://github.com/Cloufield/gwaslab/blob/main/examples/7_visualization/visualization_regional.ipynb">gwaslab/examples/7_visualization/visualization_regional.ipynb at main · Cloufield/gwaslab</a></p>
<figure class="highlight plaintext"><table><tr><td class="code"><pre><span class="line"># 创建新环境</span><br><span class="line"># conda env create -n gwaslab -c conda-forge python=3.9</span><br><span class="line"># pip install gwaslab</span><br><span class="line"># pip install wget</span><br></pre></td></tr></table></figure>
<figure class="highlight python"><table><tr><td class="code"><pre><span class="line"><span class="keyword">import</span> wget</span><br><span class="line"><span class="keyword">import</span> inspect</span><br><span class="line"><span class="keyword">import</span> sys</span><br><span class="line"><span class="keyword">import</span> gwaslab <span class="keyword">as</span> gl</span><br><span class="line"></span><br><span class="line"><span class="comment">########### 0. 下载（加载）(示例)数据 ###########</span></span><br><span class="line"><span class="comment"># download</span></span><br><span class="line">wget.download(<span class="string">&quot;http://jenger.riken.jp/14/&quot;</span>, out = <span class="string">&quot;/data2/projects/bioinfo/tyliu/1_proteomics/MR/ckb_mr/coloc/plot/exam/t2d_bbj.txt.gz&quot;</span>)</span><br><span class="line"></span><br><span class="line"><span class="comment"># load</span></span><br><span class="line">mysumstats = gl.Sumstats(<span class="string">&quot;/data2/projects/bioinfo/tyliu/1_proteomics/MR/ckb_mr/coloc/plot/exam/t2d_bbj.txt.gz&quot;</span>,</span><br><span class="line">             snpid = <span class="string">&quot;SNP&quot;</span>,</span><br><span class="line">             chrom = <span class="string">&quot;CHR&quot;</span>,</span><br><span class="line">             pos = <span class="string">&quot;POS&quot;</span>,</span><br><span class="line">             ea = <span class="string">&quot;ALT&quot;</span>,</span><br><span class="line">             nea = <span class="string">&quot;REF&quot;</span>,</span><br><span class="line">             neaf = <span class="string">&quot;Frq&quot;</span>,</span><br><span class="line">             beta = <span class="string">&quot;BETA&quot;</span>,</span><br><span class="line">             se = <span class="string">&quot;SE&quot;</span>,</span><br><span class="line">             p = <span class="string">&quot;P&quot;</span>,</span><br><span class="line">             direction = <span class="string">&quot;Dir&quot;</span>,</span><br><span class="line">             build = <span class="string">&quot;19&quot;</span>, <span class="comment"># hg19/38, 更换为38可行</span></span><br><span class="line">             chrom_pat = <span class="string">&quot;7&quot;</span>, <span class="comment"># 仅加载7号染色体</span></span><br><span class="line">             n = <span class="string">&quot;N&quot;</span>)</span><br><span class="line">mysumstats.basic_check(verbose = <span class="literal">False</span>) </span><br><span class="line"></span><br><span class="line"><span class="comment"># print(inspect.getsource(mysumstats.basic_check)) # 函数源代码</span></span><br><span class="line"><span class="comment"># 检查SNP格式；检查CHR列是否输入正确；检查POS是否准确；等位基因是否标记正确；效应值、标准误、P值之间数量关系的检查；</span></span><br></pre></td></tr></table></figure>
<h2 id="1-0-可用文档"><a href="#1-0-可用文档" class="headerlink" title="1.0 可用文档"></a>1.0 可用文档</h2><figure class="highlight python"><table><tr><td class="code"><pre><span class="line">gl.check_available_ref()</span><br></pre></td></tr></table></figure>
<h2 id="1-1-曼哈顿图"><a href="#1-1-曼哈顿图" class="headerlink" title="1.1 曼哈顿图"></a>1.1 曼哈顿图</h2><figure class="highlight python"><table><tr><td class="code"><pre><span class="line"><span class="comment">########### 1. 使用Summary data绘制曼哈顿图（一条染色体） ###########</span></span><br><span class="line"><span class="comment"># print(inspect.getsource(mysumstats.plot_mqq))</span></span><br><span class="line">mqq_fig, mqq_ax = mysumstats.plot_mqq(</span><br><span class="line">    mode=<span class="string">&quot;m&quot;</span>,</span><br><span class="line">    anno=<span class="string">&quot;GENENAME&quot;</span>,</span><br><span class="line">    anno_source=<span class="string">&quot;ensembl&quot;</span></span><br><span class="line">)</span><br><span class="line">mqq_fig.savefig(<span class="string">&quot;/data2/projects/bioinfo/tyliu/1_proteomics/MR/ckb_mr/coloc/plot/exam/manhattan_plot.png&quot;</span>)</span><br></pre></td></tr></table></figure>
<p><img src="/img/loading.gif" data-original="/2025/03/19/Python-StudyNotes-GWASlab/manhattan_plot%20%5B2%5D.png" alt="基础曼哈顿图"></p>
<h2 id="1-2-区域图"><a href="#1-2-区域图" class="headerlink" title="1.2 区域图"></a>1.2 区域图</h2><figure class="highlight python"><table><tr><td class="code"><pre><span class="line"><span class="comment"># https://cloufield.github.io/gwaslab/RegionalPlot/</span></span><br><span class="line"></span><br><span class="line"><span class="comment">########### 2. 识别lead snp ###########</span></span><br><span class="line"><span class="comment"># 滑动窗口500Kb识别lead snp，阈值5e-8</span></span><br><span class="line">mysumstats.get_lead()</span><br><span class="line"></span><br><span class="line"><span class="comment">########### 3. 绘制指定区域Regional plot ###########</span></span><br><span class="line">region_fig, region_ax = mysumstats.plot_mqq(region=(<span class="number">7</span>,<span class="number">156538803</span>,<span class="number">157538803</span>))</span><br><span class="line"></span><br><span class="line">region_fig.savefig(<span class="string">&quot;/data2/projects/bioinfo/tyliu/1_proteomics/MR/ckb_mr/coloc/plot/exam/region_base_plot.png&quot;</span>)</span><br></pre></td></tr></table></figure>
<p><img src="/img/loading.gif" data-original="/2025/03/19/Python-StudyNotes-GWASlab/region_base_plot.png" alt="基础区域图"></p>
<p>左图是简单的区域图，右图是Q-Q图，用于评估p值分布的合理性。图中$\lambda_{GC}$（<strong>基因组膨胀因子</strong>）用于衡量统计量的潜在膨胀或收缩。此处Q-Q图绘制的目标是观察到的p值分布与理论期望分布（假设无真实关联时的随机分布）。</p>
<ul>
<li>X轴：理论期望的p值，即均匀分布下的p值；</li>
<li>Y轴：实际分布得到的p值；</li>
<li>理想情况：所有点沿着对角线分布；</li>
<li>偏离情况：1、整体高于对角线：此时存在真实关联信号或系统性误差（群体分层、技术偏差）；2、整体低于对角线：可能存在过校正或统计效力不足的问题。</li>
</ul>
<p>$\lambda_{GC}=\frac{\text{观察到的卡方统计量中位数}}{\text{理论卡方分布中位数(}0.456\mathrm{)}}$，当$\lambda_{GC}&gt;1$时存在系统性膨胀；$\lambda_{GC}&lt;1$时表示统计效力不足；$\lambda_{GC} \approx 1$时表示符合期望。</p>
<p>整体观察左右两图：</p>
<ol>
<li>若区域图中有显著峰，并且Q-Q图中存在点明显偏离对角线，说明信息真实；</li>
<li>若Q-Q图整体膨胀，区域图中的峰可能包含假阳性。</li>
</ol>
<figure class="highlight python"><table><tr><td class="code"><pre><span class="line"><span class="comment">########### 4. 绘制带有基因轨迹的Regional plot ###########</span></span><br><span class="line"><span class="comment"># https://github.com/Cloufield/gwaslab/blob/d2e4e4b88ae8b75a3416f622d1de657cef966ed1/src/gwaslab/viz_plot_mqqplot.py#L80C1-L84C26</span></span><br><span class="line"><span class="comment"># 指定Recombination的路径</span></span><br><span class="line"><span class="comment"># print(inspect.getsource(mysumstats.plot_mqq))</span></span><br><span class="line"><span class="comment"># 设置数据路径</span></span><br><span class="line">region_withTRACE_fig, region_withTRACE_ax = mysumstats.plot_mqq(mode=<span class="string">&quot;r&quot;</span>,region=(<span class="number">7</span>,<span class="number">156538803</span>,<span class="number">157538803</span>),region_grid=<span class="literal">True</span>, rr_path=<span class="string">&quot;/data2/projects/bioinfo/tyliu/1_proteomics/MR/ckb_mr/coloc/plot/exam/recombination/hg19/genetic_map_GRCh37_chr7.txt&quot;</span>,</span><br><span class="line">                                           rr_header_dict=<span class="literal">None</span>,</span><br><span class="line">                                           rr_chr_dict = <span class="literal">None</span>,</span><br><span class="line">                                           rr_lim=(<span class="number">0</span>,<span class="number">100</span>),</span><br><span class="line">                                           rr_ylabel=<span class="literal">True</span>)</span><br><span class="line"></span><br><span class="line">region_withTRACE_fig.savefig(<span class="string">&quot;/data2/projects/bioinfo/tyliu/1_proteomics/MR/ckb_mr/coloc/plot/exam/region_withTRACE_plot.png&quot;</span>)</span><br></pre></td></tr></table></figure>
<p><img src="/img/loading.gif" data-original="/2025/03/19/Python-StudyNotes-GWASlab/image-20250327115051155.png" alt="基因轨迹区域图"></p>
<figure class="highlight r"><table><tr><td class="code"><pre><span class="line"><span class="comment">########### 5. 绘制带有基因轨迹和LD information的Regional plot ###########</span></span><br><span class="line">local_vcf_path <span class="operator">=</span> <span class="string">&quot;/data2/projects/bioinfo/tyliu/1_proteomics/MR/ckb_mr/coloc/plot/exam/1k_eas_hg19/EAS.ALL.split_norm_af.1kgp3v5.hg19.vcf.gz&quot;</span></span><br><span class="line"></span><br><span class="line">region_withTRACE_LD_fig<span class="punctuation">,</span> region_withTRACE_LD_ax <span class="operator">=</span> mysumstats.plot_mqq<span class="punctuation">(</span>mode<span class="operator">=</span><span class="string">&quot;r&quot;</span><span class="punctuation">,</span>region<span class="operator">=</span><span class="punctuation">(</span><span class="number">7</span><span class="punctuation">,</span><span class="number">156538803</span><span class="punctuation">,</span><span class="number">157538803</span><span class="punctuation">)</span><span class="punctuation">,</span>region_grid<span class="operator">=</span>True<span class="punctuation">,</span> rr_path<span class="operator">=</span><span class="string">&quot;/data2/projects/bioinfo/tyliu/1_proteomics/MR/ckb_mr/coloc/plot/exam/recombination/hg19/genetic_map_GRCh37_chr7.txt&quot;</span><span class="punctuation">,</span></span><br><span class="line">                                           vcf_path <span class="operator">=</span> local_vcf_path<span class="punctuation">)</span></span><br><span class="line"></span><br><span class="line">region_withTRACE_LD_fig.savefig<span class="punctuation">(</span><span class="string">&quot;/data2/projects/bioinfo/tyliu/1_proteomics/MR/ckb_mr/coloc/plot/exam/region_withTRACE_LD_plot.png&quot;</span><span class="punctuation">)</span></span><br></pre></td></tr></table></figure>
<p><img src="/img/loading.gif" data-original="/2025/03/19/Python-StudyNotes-GWASlab/region_withTRACE_LD_plot.png" alt="进阶基因区域图"></p>
<figure class="highlight r"><table><tr><td class="code"><pre><span class="line"><span class="comment">########### 6. Stacked Regional plot ###########</span></span><br><span class="line">local_vcf_path_eas <span class="operator">=</span> <span class="string">&quot;/data2/projects/bioinfo/tyliu/1_proteomics/MR/ckb_mr/coloc/plot/exam/1k_eas_hg19/EAS.ALL.split_norm_af.1kgp3v5.hg19.vcf.gz&quot;</span></span><br><span class="line">local_vcf_path_eur <span class="operator">=</span> <span class="string">&quot;/data2/projects/bioinfo/tyliu/1_proteomics/MR/ckb_mr/coloc/plot/exam/1k_eur_hg19/EUR.ALL.split_norm_af.1kgp3v5.hg19.vcf.gz&quot;</span></span><br><span class="line"></span><br><span class="line">stacked_region_fig<span class="punctuation">,</span> stacked_region_ax <span class="operator">=</span>  gl.plot_stacked_mqq<span class="punctuation">(</span>objects<span class="operator">=</span><span class="punctuation">[</span>mysumstats<span class="punctuation">,</span>mysumstats<span class="punctuation">]</span><span class="punctuation">,</span></span><br><span class="line">                    vcfs<span class="operator">=</span><span class="punctuation">[</span>local_vcf_path_eas<span class="punctuation">,</span> local_vcf_path_eur<span class="punctuation">]</span><span class="punctuation">,</span></span><br><span class="line">                    region<span class="operator">=</span><span class="punctuation">(</span><span class="number">7</span><span class="punctuation">,</span><span class="number">156538803</span><span class="punctuation">,</span><span class="number">157538803</span><span class="punctuation">)</span><span class="punctuation">,</span> </span><br><span class="line">                    mode<span class="operator">=</span><span class="string">&quot;r&quot;</span><span class="punctuation">,</span></span><br><span class="line">                    build<span class="operator">=</span><span class="string">&quot;19&quot;</span><span class="punctuation">,</span> rr_path<span class="operator">=</span><span class="string">&quot;/data2/projects/bioinfo/tyliu/1_proteomics/MR/ckb_mr/coloc/plot/exam/recombination/hg19/genetic_map_GRCh37_chr7.txt&quot;</span><span class="punctuation">,</span></span><br><span class="line">                    anno<span class="operator">=</span>True<span class="punctuation">,</span></span><br><span class="line">                    anno_style<span class="operator">=</span><span class="string">&quot;right&quot;</span><span class="punctuation">,</span>anno_xshift<span class="operator">=</span><span class="number">0.02</span><span class="punctuation">,</span></span><br><span class="line">                    anno_set<span class="operator">=</span><span class="punctuation">[</span><span class="string">&quot;7:156793450_G_GA&quot;</span><span class="punctuation">,</span><span class="string">&quot;7:157038803_A_G&quot;</span><span class="punctuation">]</span><span class="punctuation">,</span></span><br><span class="line">                    titles<span class="operator">=</span><span class="punctuation">[</span><span class="string">&quot;EAS&quot;</span><span class="punctuation">,</span><span class="string">&quot;EUR&quot;</span><span class="punctuation">]</span><span class="punctuation">,</span></span><br><span class="line">                    title_args<span class="operator">=</span><span class="punctuation">&#123;</span><span class="string">&quot;size&quot;</span><span class="operator">:</span><span class="number">20</span><span class="punctuation">&#125;</span><span class="punctuation">,</span></span><br><span class="line">                    anno_args<span class="operator">=</span><span class="punctuation">&#123;</span><span class="string">&quot;rotation&quot;</span><span class="operator">:</span><span class="number">0</span><span class="punctuation">&#125;</span><span class="punctuation">)</span></span><br><span class="line">stacked_region_fig.savefig<span class="punctuation">(</span><span class="string">&quot;/data2/projects/bioinfo/tyliu/1_proteomics/MR/ckb_mr/coloc/plot/exam/stacked_plot.png&quot;</span><span class="punctuation">)</span></span><br></pre></td></tr></table></figure>
<p><img src="/img/loading.gif" data-original="/2025/03/19/Python-StudyNotes-GWASlab/image-20250328101231135.png" alt="stack"></p>
<p>注：GWAS数据本身是EAS的数据，所以这里只是函数功能的展示，并不可以使用EUR的Reference进行LD信息的参考。</p>
</article><div class="post-copyright"><div class="post-copyright__author"><span class="post-copyright-meta"><i class="fas fa-circle-user fa-fw"></i>文章作者: </span><span class="post-copyright-info"><a href="http://example.com">远山</a></span></div><div class="post-copyright__type"><span class="post-copyright-meta"><i class="fas fa-square-arrow-up-right fa-fw"></i>文章链接: </span><span class="post-copyright-info"><a href="http://example.com/2025/03/19/Python-StudyNotes-GWASlab/">http://example.com/2025/03/19/Python-StudyNotes-GWASlab/</a></span></div><div class="post-copyright__notice"><span class="post-copyright-meta"><i class="fas fa-circle-exclamation fa-fw"></i>版权声明: </span><span class="post-copyright-info">本博客所有文章除特别声明外，均采用 <a href="https://creativecommons.org/licenses/by-nc-sa/4.0/" target="_blank">CC BY-NC-SA 4.0</a> 许可协议。转载请注明来自 <a href="http://example.com" target="_blank">FarMountBiolog</a>！</span></div></div><div class="tag_share"><div class="post-meta__tag-list"><a class="post-meta__tags" href="/tags/Pythony/">Pythony</a></div><div class="post_share"><div class="social-share" data-image="/img/Kakasi.jpg" data-sites="facebook,twitter,wechat,weibo,qq"></div><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/butterfly-extsrc@1.1.3/sharejs/dist/css/share.min.css" media="print" onload="this.media='all'"><script src="https://cdn.jsdelivr.net/npm/butterfly-extsrc@1.1.3/sharejs/dist/js/social-share.min.js" defer></script></div></div><nav class="pagination-post" id="pagination"><div class="prev-post pull-left"><a href="/2025/03/28/R-StudyNotes-locuszoomr/" title="R-StudyNotes-locuszoomr"><div class="cover" style="background: var(--default-bg-color)"></div><div class="pagination-info"><div class="label">上一篇</div><div class="prev_info">R-StudyNotes-locuszoomr</div></div></a></div><div class="next-post pull-right"><a href="/2025/01/03/Genomics/" title="Genomic"><div class="cover" style="background: var(--default-bg-color)"></div><div class="pagination-info"><div class="label">下一篇</div><div class="next_info">Genomic</div></div></a></div></nav><hr class="custom-hr"/><div id="post-comment"><div class="comment-head"><div class="comment-headline"><i class="fas fa-comments fa-fw"></i><span> 评论</span></div></div><div class="comment-wrap"><div><div id="gitalk-container"></div></div></div></div></div><div class="aside-content" id="aside-content"><div class="card-widget card-info"><div class="is-center"><div class="avatar-img"><img src="/img/loading.gif" data-original="/img/Kakasi.jpg" onerror="this.onerror=null;this.src='/img/friend_404.gif'" alt="avatar"/></div><div class="author-info__name">远山</div><div class="author-info__description">大鹏一日同风起，扶摇直上九万里</div></div><div class="card-info-data site-data is-center"><a href="/archives/"><div class="headline">文章</div><div class="length-num">10</div></a><a href="/tags/"><div class="headline">标签</div><div class="length-num">4</div></a><a href="/categories/"><div class="headline">分类</div><div class="length-num">0</div></a></div><a id="card-info-btn" target="_blank" rel="noopener" href="https://github.com/BenjaminMount"><i class="fab fa-github"></i><span>Follow Me</span></a><div class="card-info-social-icons is-center"><a class="social-icon" href="https://github.com/BenjaminMount/" target="_blank" title="Github"><i class="fab fa-github"></i></a><a class="social-icon" href="tencent://AddContact/?fromId=45&amp;fromSubId=1&amp;subcmd=all&amp;uin=2396979300&amp;website=www.oicqzone.com" target="_blank" title="QQ"><i class="fab fa-qq"></i></a><a class="social-icon" href="mailto:liutongyan@hust.edu.cn" target="_blank" title="Email"><i class="fas fa-envelope-open-text"></i></a></div></div><div class="card-widget card-announcement"><div class="item-headline"><i class="fas fa-bullhorn fa-shake"></i><span>公告</span></div><div class="announcement_content">Keep thinking. Progress itself is hard.</div></div><div class="sticky_layout"><div class="card-widget" id="card-toc"><div class="item-headline"><i class="fas fa-stream"></i><span>目录</span><span class="toc-percentage"></span></div><div class="toc-content is-expand"><ol class="toc"><li class="toc-item toc-level-1"><a class="toc-link" href="#1-GWAS%E5%8F%AF%E8%A7%86%E5%8C%96%E5%AD%A6%E4%B9%A0"><span class="toc-text">1. GWAS可视化学习</span></a><ol class="toc-child"><li class="toc-item toc-level-2"><a class="toc-link" href="#1-0-%E5%8F%AF%E7%94%A8%E6%96%87%E6%A1%A3"><span class="toc-text">1.0 可用文档</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#1-1-%E6%9B%BC%E5%93%88%E9%A1%BF%E5%9B%BE"><span class="toc-text">1.1 曼哈顿图</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#1-2-%E5%8C%BA%E5%9F%9F%E5%9B%BE"><span class="toc-text">1.2 区域图</span></a></li></ol></li></ol></div></div><div class="card-widget card-recent-post"><div class="item-headline"><i class="fas fa-history"></i><span>最新文章</span></div><div class="aside-list"><div class="aside-list-item no-cover"><div class="content"><a class="title" href="/2025/04/09/EfficiencyStandError/" title="Efficiency and StandError">Efficiency and StandError</a><time datetime="2025-04-09T04:00:06.000Z" title="发表于 2025-04-09 12:00:06">2025-04-09</time></div></div><div class="aside-list-item no-cover"><div class="content"><a class="title" href="/2025/03/28/R-StudyNotes-locuszoomr/" title="R-StudyNotes-locuszoomr">R-StudyNotes-locuszoomr</a><time datetime="2025-03-28T08:15:12.000Z" title="发表于 2025-03-28 16:15:12">2025-03-28</time></div></div><div class="aside-list-item no-cover"><div class="content"><a class="title" href="/2025/03/19/Python-StudyNotes-GWASlab/" title="Python-StudyNotes-GWASlab">Python-StudyNotes-GWASlab</a><time datetime="2025-03-19T14:00:24.000Z" title="发表于 2025-03-19 22:00:24">2025-03-19</time></div></div><div class="aside-list-item no-cover"><div class="content"><a class="title" href="/2025/01/03/Genomics/" title="Genomic">Genomic</a><time datetime="2025-01-03T07:55:58.000Z" title="发表于 2025-01-03 15:55:58">2025-01-03</time></div></div><div class="aside-list-item no-cover"><div class="content"><a class="title" href="/2024/10/14/Basic%20statistic%20method/" title="Basic Statistic Method">Basic Statistic Method</a><time datetime="2024-10-14T06:52:21.000Z" title="发表于 2024-10-14 14:52:21">2024-10-14</time></div></div></div></div></div></div></main><footer id="footer" style="background-image: url('/img/pic_1.png')"><div id="footer-wrap"><div class="copyright">&copy;2024 - 2025 By 远山</div><div class="framework-info"><span>框架 </span><a target="_blank" rel="noopener" href="https://hexo.io">Hexo</a><span class="footer-separator">|</span><span>主题 </span><a target="_blank" rel="noopener" href="https://github.com/jerryc127/hexo-theme-butterfly">Butterfly</a></div></div></footer></div><div id="rightside"><div id="rightside-config-hide"><button id="readmode" type="button" title="阅读模式"><i class="fas fa-book-open"></i></button><button id="darkmode" type="button" title="浅色和深色模式转换"><i class="fas fa-adjust"></i></button><button id="hide-aside-btn" type="button" title="单栏和双栏切换"><i class="fas fa-arrows-alt-h"></i></button></div><div id="rightside-config-show"><button id="rightside-config" type="button" title="设置"><i class="fas fa-cog fa-spin"></i></button><button class="close" id="mobile-toc-button" type="button" title="目录"><i class="fas fa-list-ul"></i></button><a id="to_comment" href="#post-comment" title="直达评论"><i class="fas fa-comments"></i></a><button id="go-up" type="button" title="回到顶部"><span class="scroll-percent"></span><i class="fas fa-arrow-up"></i></button></div></div><div><script src="/js/utils.js?v=4.13.0"></script><script src="/js/main.js?v=4.13.0"></script><script src="https://cdn.jsdelivr.net/npm/medium-zoom@1.1.0/dist/medium-zoom.min.js"></script><div class="js-pjax"><script>if (!window.MathJax) {
  window.MathJax = {
    tex: {
      inlineMath: [['$', '$'], ['\\(', '\\)']],
      tags: 'ams'
    },
    chtml: {
      scale: 1.1
    },
    options: {
      renderActions: {
        findScript: [10, doc => {
          for (const node of document.querySelectorAll('script[type^="math/tex"]')) {
            const display = !!node.type.match(/; *mode=display/)
            const math = new doc.options.MathItem(node.textContent, doc.inputJax[0], display)
            const text = document.createTextNode('')
            node.parentNode.replaceChild(text, node)
            math.start = {node: text, delim: '', n: 0}
            math.end = {node: text, delim: '', n: 0}
            doc.math.push(math)
          }
        }, '']
      }
    }
  }
  
  const script = document.createElement('script')
  script.src = 'https://cdn.jsdelivr.net/npm/mathjax@3.2.2/es5/tex-mml-chtml.min.js'
  script.id = 'MathJax-script'
  script.async = true
  document.head.appendChild(script)
} else {
  MathJax.startup.document.state(0)
  MathJax.texReset()
  MathJax.typesetPromise()
}</script><script>(() => {
  const initGitalk = () => {
    const gitalk = new Gitalk(Object.assign({
      clientID: 'Ov23liSwPtc8jModvEnf',
      clientSecret: '087679c1568bceccc832c27ef3c219a01cb93a8e',
      repo: 'BenjaminMount.github.io',
      owner: 'BenjaminMount',
      admin: ['BenjaminMount'],
      id: 'e5ea93ff7d2bb617289755247961bba0',
      updateCountCallback: commentCount
    },null))

    gitalk.render('gitalk-container')
  }

  const loadGitalk = async() => {
    if (typeof Gitalk === 'function') initGitalk()
    else {
      await getCSS('https://cdn.jsdelivr.net/npm/gitalk@1.8.0/dist/gitalk.min.css')
      await getScript('https://cdn.jsdelivr.net/npm/gitalk@1.8.0/dist/gitalk.min.js')
      initGitalk()
    }
  }
  
  const commentCount = n => {
    const isCommentCount = document.querySelector('#post-meta .gitalk-comment-count')
    if (isCommentCount) {
      isCommentCount.textContent= n
    }
  }

  if ('Gitalk' === 'Gitalk' || !true) {
    if (true) btf.loadComment(document.getElementById('gitalk-container'), loadGitalk)
    else loadGitalk()
  } else {
    window.loadOtherComment = loadGitalk
  }
})()</script></div><script id="canvas_nest" defer="defer" color="0,0,255" opacity="0.7" zIndex="-1" count="99" mobile="false" src="https://cdn.jsdelivr.net/npm/butterfly-extsrc@1.1.3/dist/canvas-nest.min.js"></script><script src="https://cdn.jsdelivr.net/npm/butterfly-extsrc@1.1.3/dist/activate-power-mode.min.js"></script><script>POWERMODE.colorful = true;
POWERMODE.shake = false;
POWERMODE.mobile = false;
document.body.addEventListener('input', POWERMODE);
</script><script async data-pjax src="//busuanzi.ibruce.info/busuanzi/2.3/busuanzi.pure.mini.js"></script><div id="local-search"><div class="search-dialog"><nav class="search-nav"><span class="search-dialog-title">搜索</span><span id="loading-status"></span><button class="search-close-button"><i class="fas fa-times"></i></button></nav><div class="is-center" id="loading-database"><i class="fas fa-spinner fa-pulse"></i><span>  数据库加载中</span></div><div class="search-wrap"><div id="local-search-input"><div class="local-search-box"><input class="local-search-box--input" placeholder="搜索文章" type="text"/></div></div><hr/><div id="local-search-results"></div><div id="local-search-stats-wrap"></div></div></div><div id="search-mask"></div><script src="/js/search/local-search.js?v=4.13.0"></script></div></div>
        <style>
            [bg-lazy] {
                background-image: none !important;
                background-color: #eee !important;
            }
        </style>
        <script>
            window.imageLazyLoadSetting = {
                isSPA: false,
                preloadRatio: 1,
                processImages: null,
            };
        </script><script>window.addEventListener("load",function(){var t=/\.(gif|jpg|jpeg|tiff|png)$/i,r=/^data:image\/[a-z]+;base64,/;Array.prototype.slice.call(document.querySelectorAll("img[data-original]")).forEach(function(a){var e=a.parentNode;"A"===e.tagName&&(e.href.match(t)||e.href.match(r))&&(e.href=a.dataset.original)})});</script><script>(r=>{r.imageLazyLoadSetting.processImages=t;var a=r.imageLazyLoadSetting.isSPA,o=r.imageLazyLoadSetting.preloadRatio||1,d=i();function i(){var t=Array.prototype.slice.call(document.querySelectorAll("img[data-original]")),e=Array.prototype.slice.call(document.querySelectorAll("[bg-lazy]"));return t.concat(e)}function t(t){(a||t)&&(d=i());for(var e,n=0;n<d.length;n++)0<=(e=(e=d[n]).getBoundingClientRect()).bottom&&0<=e.left&&e.top<=(r.innerHeight*o||document.documentElement.clientHeight*o)&&(()=>{var t,e,a,o,i=d[n];e=function(){d=d.filter(function(t){return i!==t}),r.imageLazyLoadSetting.onImageLoaded&&r.imageLazyLoadSetting.onImageLoaded(i)},(t=i).dataset.loaded||(t.hasAttribute("bg-lazy")?(t.removeAttribute("bg-lazy"),e&&e()):(a=new Image,o=t.getAttribute("data-original"),a.onload=function(){t.src=o,t.removeAttribute("data-original"),t.setAttribute("data-loaded",!0),e&&e()},a.onerror=function(){t.removeAttribute("data-original"),t.setAttribute("data-loaded",!1),t.src=o},t.src!==o&&(a.src=o)))})()}function e(){clearTimeout(t.tId),t.tId=setTimeout(t,500)}t(),document.addEventListener("scroll",e),r.addEventListener("resize",e),r.addEventListener("orientationchange",e)})(this);</script></body></html>