<!DOCTYPE html><html lang="zh-CN" data-theme="light"><head><meta charset="UTF-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0,viewport-fit=cover"><title>Proteomics | FarMountBiolog</title><meta name="author" content="远山"><meta name="copyright" content="远山"><meta name="format-detection" content="telephone=no"><meta name="theme-color" content="#ffffff"><meta name="description" content="Proteomics0 Preface蛋白质组学分析基础概念知识。 蛋白质组学研究蛋白质结构、功能及其在生物系统中表达。分析过程中通常需要处理大量复杂的数据，包括蛋白质的鉴定、定量、修饰、相互作用等。 一般流程包括： 数据获取与预处理 蛋白质鉴定：利用质谱（Mass Spectrometry，MS）技术识别样本中的蛋白质。 原始数据处理：将质谱数据转化为可识别的肽段和蛋白质信息。 数据过滤：移除低">
<meta property="og:type" content="article">
<meta property="og:title" content="Proteomics">
<meta property="og:url" content="http://example.com/2024/08/26/proteomics/index.html">
<meta property="og:site_name" content="FarMountBiolog">
<meta property="og:description" content="Proteomics0 Preface蛋白质组学分析基础概念知识。 蛋白质组学研究蛋白质结构、功能及其在生物系统中表达。分析过程中通常需要处理大量复杂的数据，包括蛋白质的鉴定、定量、修饰、相互作用等。 一般流程包括： 数据获取与预处理 蛋白质鉴定：利用质谱（Mass Spectrometry，MS）技术识别样本中的蛋白质。 原始数据处理：将质谱数据转化为可识别的肽段和蛋白质信息。 数据过滤：移除低">
<meta property="og:locale" content="zh_CN">
<meta property="og:image" content="http://example.com/img/Kakasi.jpg">
<meta property="article:published_time" content="2024-08-26T03:48:19.000Z">
<meta property="article:modified_time" content="2024-09-03T08:11:26.977Z">
<meta property="article:author" content="远山">
<meta property="article:tag" content="StudyNotes">
<meta name="twitter:card" content="summary">
<meta name="twitter:image" content="http://example.com/img/Kakasi.jpg"><link rel="shortcut icon" href="/img/Celtics.jpg"><link rel="canonical" href="http://example.com/2024/08/26/proteomics/index.html"><link rel="preconnect" href="//cdn.jsdelivr.net"/><link rel="preconnect" href="//busuanzi.ibruce.info"/><link rel="stylesheet" href="/css/index.css?v=4.13.0"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fortawesome/fontawesome-free@6.5.1/css/all.min.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fancyapps/ui@5.0.33/dist/fancybox/fancybox.min.css" media="print" onload="this.media='all'"><script>const GLOBAL_CONFIG = {
  root: '/',
  algolia: undefined,
  localSearch: {"path":"/search.xml","preload":false,"top_n_per_article":1,"unescape":false,"languages":{"hits_empty":"找不到您查询的内容：${query}","hits_stats":"共找到 ${hits} 篇文章"}},
  translate: undefined,
  noticeOutdate: undefined,
  highlight: {"plugin":"highlight.js","highlightCopy":true,"highlightLang":false,"highlightHeightLimit":false},
  copy: {
    success: '复制成功',
    error: '复制错误',
    noSupport: '浏览器不支持'
  },
  relativeDate: {
    homepage: false,
    post: false
  },
  runtime: '',
  dateSuffix: {
    just: '刚刚',
    min: '分钟前',
    hour: '小时前',
    day: '天前',
    month: '个月前'
  },
  copyright: undefined,
  lightbox: 'mediumZoom',
  Snackbar: undefined,
  infinitegrid: {
    js: 'https://cdn.jsdelivr.net/npm/@egjs/infinitegrid@4.11.1/dist/infinitegrid.min.js',
    buttonText: '加载更多'
  },
  isPhotoFigcaption: true,
  islazyload: false,
  isAnchor: false,
  percent: {
    toc: true,
    rightside: false,
  },
  autoDarkmode: false
}</script><script id="config-diff">var GLOBAL_CONFIG_SITE = {
  title: 'Proteomics',
  isPost: true,
  isHome: false,
  isHighlightShrink: false,
  isToc: true,
  postUpdate: '2024-09-03 16:11:26'
}</script><script>(win=>{
      win.saveToLocal = {
        set: (key, value, ttl) => {
          if (ttl === 0) return
          const now = Date.now()
          const expiry = now + ttl * 86400000
          const item = {
            value,
            expiry
          }
          localStorage.setItem(key, JSON.stringify(item))
        },
      
        get: key => {
          const itemStr = localStorage.getItem(key)
      
          if (!itemStr) {
            return undefined
          }
          const item = JSON.parse(itemStr)
          const now = Date.now()
      
          if (now > item.expiry) {
            localStorage.removeItem(key)
            return undefined
          }
          return item.value
        }
      }
    
      win.getScript = (url, attr = {}) => new Promise((resolve, reject) => {
        const script = document.createElement('script')
        script.src = url
        script.async = true
        script.onerror = reject
        script.onload = script.onreadystatechange = function() {
          const loadState = this.readyState
          if (loadState && loadState !== 'loaded' && loadState !== 'complete') return
          script.onload = script.onreadystatechange = null
          resolve()
        }

        Object.keys(attr).forEach(key => {
          script.setAttribute(key, attr[key])
        })

        document.head.appendChild(script)
      })
    
      win.getCSS = (url, id = false) => new Promise((resolve, reject) => {
        const link = document.createElement('link')
        link.rel = 'stylesheet'
        link.href = url
        if (id) link.id = id
        link.onerror = reject
        link.onload = link.onreadystatechange = function() {
          const loadState = this.readyState
          if (loadState && loadState !== 'loaded' && loadState !== 'complete') return
          link.onload = link.onreadystatechange = null
          resolve()
        }
        document.head.appendChild(link)
      })
    
      win.activateDarkMode = () => {
        document.documentElement.setAttribute('data-theme', 'dark')
        if (document.querySelector('meta[name="theme-color"]') !== null) {
          document.querySelector('meta[name="theme-color"]').setAttribute('content', '#0d0d0d')
        }
      }
      win.activateLightMode = () => {
        document.documentElement.setAttribute('data-theme', 'light')
        if (document.querySelector('meta[name="theme-color"]') !== null) {
          document.querySelector('meta[name="theme-color"]').setAttribute('content', '#ffffff')
        }
      }
      const t = saveToLocal.get('theme')
    
        if (t === 'dark') activateDarkMode()
        else if (t === 'light') activateLightMode()
      
      const asideStatus = saveToLocal.get('aside-status')
      if (asideStatus !== undefined) {
        if (asideStatus === 'hide') {
          document.documentElement.classList.add('hide-aside')
        } else {
          document.documentElement.classList.remove('hide-aside')
        }
      }
    
      const detectApple = () => {
        if(/iPad|iPhone|iPod|Macintosh/.test(navigator.userAgent)){
          document.documentElement.classList.add('apple')
        }
      }
      detectApple()
    })(window)</script><meta name="generator" content="Hexo 7.3.0"></head><body><div id="sidebar"><div id="menu-mask"></div><div id="sidebar-menus"><div class="avatar-img is-center"><img src="/img/loading.gif" data-original="/img/Kakasi.jpg" onerror="onerror=null;src='/img/friend_404.gif'" alt="avatar"/></div><div class="sidebar-site-data site-data is-center"><a href="/archives/"><div class="headline">文章</div><div class="length-num">4</div></a><a href="/tags/"><div class="headline">标签</div><div class="length-num">2</div></a><a href="/categories/"><div class="headline">分类</div><div class="length-num">0</div></a></div><hr class="custom-hr"/><div class="menus_items"><div class="menus_item"><a class="site-page" href="/"><i class="fa-fw fas fa-home"></i><span> 主页</span></a></div><div class="menus_item"><a class="site-page group" href="javascript:void(0);"><i class="fa-fw fa fa-graduation-cap"></i><span> 博文</span><i class="fas fa-chevron-down"></i></a><ul class="menus_item_child"><li><a class="site-page child" href="/archives/"><i class="fa-fw fa fa-folder-open"></i><span> 归档</span></a></li></ul></div><div class="menus_item"><a class="site-page" href="/comment/"><i class="fa-fw fa fa-paper-plane"></i><span> 留言板</span></a></div><div class="menus_item"><a class="site-page" href="/about/"><i class="fa-fw fas fa-heart"></i><span> 关于笔者</span></a></div></div></div></div><div class="post" id="body-wrap"><header class="post-bg" id="page-header" style="background-image: url('/img/pic_1.png')"><nav id="nav"><span id="blog-info"><a href="/" title="FarMountBiolog"><span class="site-name">FarMountBiolog</span></a></span><div id="menus"><div id="search-button"><a class="site-page social-icon search" href="javascript:void(0);"><i class="fas fa-search fa-fw"></i><span> 搜索</span></a></div><div class="menus_items"><div class="menus_item"><a class="site-page" href="/"><i class="fa-fw fas fa-home"></i><span> 主页</span></a></div><div class="menus_item"><a class="site-page group" href="javascript:void(0);"><i class="fa-fw fa fa-graduation-cap"></i><span> 博文</span><i class="fas fa-chevron-down"></i></a><ul class="menus_item_child"><li><a class="site-page child" href="/archives/"><i class="fa-fw fa fa-folder-open"></i><span> 归档</span></a></li></ul></div><div class="menus_item"><a class="site-page" href="/comment/"><i class="fa-fw fa fa-paper-plane"></i><span> 留言板</span></a></div><div class="menus_item"><a class="site-page" href="/about/"><i class="fa-fw fas fa-heart"></i><span> 关于笔者</span></a></div></div><div id="toggle-menu"><a class="site-page" href="javascript:void(0);"><i class="fas fa-bars fa-fw"></i></a></div></div></nav><div id="post-info"><h1 class="post-title">Proteomics</h1><div id="post-meta"><div class="meta-firstline"><span class="post-meta-date"><i class="far fa-calendar-alt fa-fw post-meta-icon"></i><span class="post-meta-label">发表于</span><time class="post-meta-date-created" datetime="2024-08-26T03:48:19.000Z" title="发表于 2024-08-26 11:48:19">2024-08-26</time><span class="post-meta-separator">|</span><i class="fas fa-history fa-fw post-meta-icon"></i><span class="post-meta-label">更新于</span><time class="post-meta-date-updated" datetime="2024-09-03T08:11:26.977Z" title="更新于 2024-09-03 16:11:26">2024-09-03</time></span></div><div class="meta-secondline"><span class="post-meta-separator">|</span><span class="post-meta-wordcount"><i class="far fa-file-word fa-fw post-meta-icon"></i><span class="post-meta-label">字数总计:</span><span class="word-count">3.4k</span><span class="post-meta-separator">|</span><i class="far fa-clock fa-fw post-meta-icon"></i><span class="post-meta-label">阅读时长:</span><span>10分钟</span></span><span class="post-meta-separator">|</span><span class="post-meta-pv-cv" id="" data-flag-title="Proteomics"><i class="far fa-eye fa-fw post-meta-icon"></i><span class="post-meta-label">阅读量:</span><span id="busuanzi_value_page_pv"><i class="fa-solid fa-spinner fa-spin"></i></span></span></div></div></div></header><main class="layout" id="content-inner"><div id="post"><article class="post-content" id="article-container"><h1 id="Proteomics"><a href="#Proteomics" class="headerlink" title="Proteomics"></a>Proteomics</h1><h2 id="0-Preface"><a href="#0-Preface" class="headerlink" title="0 Preface"></a>0 Preface</h2><p>蛋白质组学分析基础概念知识。</p>
<p>蛋白质组学研究蛋白质结构、功能及其在生物系统中表达。分析过程中通常需要处理大量复杂的数据，包括蛋白质的鉴定、定量、修饰、相互作用等。</p>
<p>一般流程包括：</p>
<h3 id="数据获取与预处理"><a href="#数据获取与预处理" class="headerlink" title="数据获取与预处理"></a>数据获取与预处理</h3><ul>
<li>蛋白质鉴定：利用质谱（Mass Spectrometry，MS）技术识别样本中的蛋白质。</li>
<li>原始数据处理：将质谱数据转化为可识别的肽段和蛋白质信息。</li>
<li>数据过滤：移除低置信度的蛋白质或肽段。</li>
</ul>
<h3 id="数据定量分析"><a href="#数据定量分析" class="headerlink" title="数据定量分析"></a>数据定量分析</h3><ul>
<li>标签定量：使用标记技术对蛋白质进行相对或绝对定量。</li>
<li>无标签定量：通过光谱计数或基于强度的方法对蛋白质定量。</li>
</ul>
<h3 id="数据标准化和归一化"><a href="#数据标准化和归一化" class="headerlink" title="数据标准化和归一化"></a>数据标准化和归一化</h3><ul>
<li>标准化：消除批次效应或实验条件的差异，常用Z-score标准化、分位数归一化。</li>
<li>归一化：调整蛋白质定量数据，使不同样本之间具有可比性。</li>
</ul>
<h3 id="差异表达分析"><a href="#差异表达分析" class="headerlink" title="差异表达分析"></a>差异表达分析</h3><ul>
<li>统计分析：使用t检验、ANOVA等识别不同条件下显著变化的蛋白质。</li>
<li>多重比较校正：校正多重比较引起的假阳性率。</li>
</ul>
<h3 id="功能注释与通路分析"><a href="#功能注释与通路分析" class="headerlink" title="功能注释与通路分析"></a>功能注释与通路分析</h3><ul>
<li>功能注释：将蛋白质映射到功能数据库理解生物学功能。</li>
<li>通路分析：利用软件识别与特定生物学通路相关的差异表达蛋白。</li>
</ul>
<h3 id="结果验证与解释"><a href="#结果验证与解释" class="headerlink" title="结果验证与解释"></a>结果验证与解释</h3><ul>
<li>验证实验：使用相关技术对关键蛋白质的表达进行验证。</li>
<li>生物学意义解释：结合实验背景，解释发现的差异蛋白质和通路的生物学意义。</li>
</ul>
<h3 id="数据可视化与报告"><a href="#数据可视化与报告" class="headerlink" title="数据可视化与报告"></a>数据可视化与报告</h3><ul>
<li>可视化：可视化展示结果。</li>
<li>撰写报告：论文撰写。</li>
</ul>
<h2 id="1-批次效应"><a href="#1-批次效应" class="headerlink" title="1 批次效应"></a>1 批次效应</h2><p><strong>批次效应</strong>：实验中不同批次而引入的系统性误差，即使在相同实验条件下，不同批次处理也可能出现系统性差异。</p>
<h3 id="影响："><a href="#影响：" class="headerlink" title="影响："></a>影响：</h3><ul>
<li>伪差异：导致样本间的差异被认为是生物学上的真实差异；</li>
<li>结果的可重复性：重复实验中可能不一致，导致数据不可重复；</li>
<li>数据分析的复杂性：数据噪音增大，掩盖了真实的生物学信号。</li>
</ul>
<h3 id="校正方法："><a href="#校正方法：" class="headerlink" title="校正方法："></a>校正方法：</h3><ul>
<li>实验设计：将不同条件下的样本随机分配到不同批次中，避免批次效应与实验因素混杂；</li>
<li>数据分析：使用统计方法校正批次效应。</li>
</ul>
<h2 id="2-多重检验校正"><a href="#2-多重检验校正" class="headerlink" title="2 多重检验校正"></a>2 多重检验校正</h2><p>当同时进行大量统计检验时，假阳性结果会大量增加，因此需要多重比较校正。</p>
<h3 id="2-1-Bonferroni"><a href="#2-1-Bonferroni" class="headerlink" title="2.1 Bonferroni"></a>2.1 Bonferroni</h3><p>最简单最严格的方法，检验N次，则将阈值设定为$P-value/N$。例如检验1000次，则将阈值设定为5%/1000=0.00005，即使检验1000次，犯错误的概率也保持在5%。</p>
<h3 id="2-2-FalseDiscovery-Rate"><a href="#2-2-FalseDiscovery-Rate" class="headerlink" title="2.2 FalseDiscovery Rate"></a>2.2 FalseDiscovery Rate</h3><p>FDR（假阳性率）法相对于Bonfer法更宽松。例如，假如检验1000次，设定阈值为5%。</p>
<p>Benjamin and Hochberg法：将P-value升序排列，计算校正后的显著性水平，对第$i$大的P-value校正：$p_i^{BH}=\frac{i}{m}\times\alpha$。$m$表示总检验数，通常$\alpha$设置为0.05（预设的显著性水平），然后将原P-value与BH:P-value相比较。</p>
<h3 id="2-3-ENT"><a href="#2-3-ENT" class="headerlink" title="2.3 ENT"></a>2.3 ENT</h3><p>Effective Number of Test是一种用于处理多重比较问题中检验之间的相关性，Bonfer直接用检验数目来作为被除数过于严格，当检验之间存在相关性时，检验数目可能高估了独立的信息量。</p>
<p>先计算所有检验统计量之间的相关矩阵，对该矩阵进行特征值分解，得到一组特征值$\lambda_1,\lambda_2,…\lambda_m$。计算ENT的值，</p>
<p>$ENT=\frac{(\sum_{i=1}^m{\lambda_i})^2}{\sum_{i=1}^m{\lambda_i^2}}$</p>
<p>校正显著性水平为：$P_{ENT}=\alpha/ENT$</p>
<p>ENT本质上是为了改善多重检验校正的结果，减少过度保守性。</p>
<h2 id="3-⭐⭐OLINK技术"><a href="#3-⭐⭐OLINK技术" class="headerlink" title="3 ⭐⭐OLINK技术"></a>3 ⭐⭐OLINK技术</h2><h3 id="3-1-OLINK"><a href="#3-1-OLINK" class="headerlink" title="3.1 OLINK"></a>3.1 OLINK</h3><p>OLINK是一项基于近邻延伸测定技术的蛋白质组学平台，通过对生物样本（血液、组织）中的蛋白质进行精准定量分析。</p>
<p>其技术的核心原理通过使用抗体对目标蛋白进行识别和绑定，每个抗体都与一个独特的DNA寡核苷酸序列连接。当两种特异性抗体与目标蛋白质结合时，它们的DNA寡核苷酸序列会靠近并形成一个可扩增的DNA模板。</p>
<p>优势：</p>
<ul>
<li>灵敏检测出低丰度蛋白质，保持高特异性；</li>
<li>多重检测能力：可以在单次实验中同时检测出多个蛋白质；</li>
<li>样本量需求低；</li>
<li>本身提供高度可重复且标准化的数据。</li>
</ul>
<h3 id="3-2-OLINK-Explore-HT"><a href="#3-2-OLINK-Explore-HT" class="headerlink" title="3.2 OLINK Explore HT"></a>3.2 OLINK Explore HT</h3><p>该技术仅需2μl样品即可检测超5300种蛋白标志物，相比于上一代Explore产品，将<strong>特异蛋白标志物检测数量提高80%</strong>，<strong>样本检测通量提高4倍</strong>，<strong>数据输出能力提高8倍</strong>，操作流程更简化，组件降低6倍，外部包装降低10倍。能够<strong>同时分析</strong>多达344个样本。</p>
<h3 id="3-3-流程设置以及操作细节"><a href="#3-3-流程设置以及操作细节" class="headerlink" title="3.3 流程设置以及操作细节"></a>3.3 流程设置以及操作细节</h3><h4 id="3-3-1-Plate"><a href="#3-3-1-Plate" class="headerlink" title="3.3.1 Plate"></a>3.3.1 Plate</h4><p>“Plate”：微孔板（microplate），用于进行高通量蛋白质分析的实验耗材，一个微孔板通常包含96孔或384孔，孔内容纳一个样本或实验反应单元。在涉及到多个目标分析下，一个Plate会进一步分为多个部分，例如“Plate 1-1”“Plate 1-2”代表同一个Plate上进行的不同批次的检测。</p>
<h4 id="3-3-2-⭐孔上外参"><a href="#3-3-2-⭐孔上外参" class="headerlink" title="3.3.2 ⭐孔上外参"></a>3.3.2 ⭐孔上外参</h4><p>孔上外参（External Control）：如：Sample Control（样本控制）、Plate Control（板内控制）、Negative Control（阴性对照）。</p>
<ul>
<li><p><strong>Sample Control</strong>：通常是已知浓度或特性的标准样本，用于验证实验过程中样本处理的正确性，确保实验过程中样本处理过程的一致性。（<strong>确保结果准确</strong>，<strong>能够正常检测</strong>）</p>
</li>
<li><p><strong>Plate Control</strong>：监控每一块板内部实验一致性，确保整个孔板的实验条件一致。（<strong>确保板内实验条件一致</strong>）</p>
</li>
<li><p><strong>Negative Control</strong>：不包含目标蛋白或抗原的对照样本，检测非特异性信号，若阴性对照中出现信号，则意味着存在非特异性，背景存在噪音。（<strong>确保无信号或信号低的蛋白不被检测</strong>）</p>
</li>
</ul>
<h4 id="3-3-3-Block稀释"><a href="#3-3-3-Block稀释" class="headerlink" title="3.3.3 Block稀释"></a>3.3.3 Block稀释</h4><p>Block通常指不同的组或者不同的板，目的是为了防止高丰度蛋白超出检测范围或低丰度蛋白无法检测，确保能够量化所有蛋白质。</p>
<h4 id="3-3-4-⭐样本内参"><a href="#3-3-4-⭐样本内参" class="headerlink" title="3.3.4 ⭐样本内参"></a>3.3.4 ⭐样本内参</h4><p>加入三种内参用于质控：</p>
<ul>
<li><p><strong>Incubation Control（孵育控制）</strong>：由一个非人类抗体组成（理论上不会与目标蛋白发生特异性结合），监控蛋白质和抗体之间的结合反应。</p>
<p>如果孵育过程是正常的，则Incubation Control的信号应当是稳定的，这是因为它本来是不会与目标蛋白结合的；<strong>若该信号低于预期，说明存在问题</strong>。</p>
</li>
<li><p><strong>Extension Control（扩展控制）</strong>：由连接在DNA标签上的抗体组成，用于检测DNA标签的扩展过程，通过检测这个信号，确保DNA扩展步骤的正常进行。<strong>若扩展控制信号异常低，意味着扩展过程发生了问题</strong>：比如在检测一种蛋白质的过程中得到了较低的扩展控制信号，这意味着在抗体扩展阶段，生成的DNA模板量少于预期，可能是DNA标签连接不充分导致了模板生成失败；可能是试剂问题；可能是孵育条件不合适等。</p>
</li>
<li><p><strong>Amplification Control（扩增控制）</strong>：通常是人工合成的双链DNA，监控扩增过程是否正常。同样，<strong>当扩增控制信号低于预期时，PCR扩增过程可能出现了问题</strong>，PCR扩增的效率会影响最终信号的强度，可能会<strong>导致蛋白的NPX值不准确</strong>。</p>
</li>
</ul>
<h3 id="3-4-OLINK-NPX数据"><a href="#3-4-OLINK-NPX数据" class="headerlink" title="3.4 OLINK NPX数据"></a>3.4 OLINK NPX数据</h3><h4 id="3-4-1-NPX数据概念"><a href="#3-4-1-NPX数据概念" class="headerlink" title="3.4.1 NPX数据概念"></a>3.4.1 NPX数据概念</h4><p>OLINK NPX（Normalized Protein eXpression）：指的是OLINK平台用于定量蛋白质表达的一种标准化数据格式（<strong>本身是一种内部标准化的过程，用于去除样本间的系统性偏差</strong>）。</p>
<script type="math/tex; mode=display">
NPX=log_2(Count^{Ext}/Count^{Ext.ctrl})</script><p>其中$Count^{Ext}$表示生成的原始荧光信号或测序数据，$Count^{Ext.ctrl}$表示扩展控制的读数，提供了一个基线信号，独立于样本中目标蛋白质浓度。通过将目标蛋白信号强度除以扩展控制信号强度，<strong>进行了标准化处理，去除样本间的系统性偏差</strong>。因此NPX整体反映了目标蛋白质的相对丰度。</p>
<h4 id="3-4-2-NPX数据标准化"><a href="#3-4-2-NPX数据标准化" class="headerlink" title="3.4.2 NPX数据标准化"></a>3.4.2 NPX数据标准化</h4><p>目的是消除系统性误差以及技术差异，使不同样本和不同板之间能够直接比较。有以下两种方案：</p>
<ul>
<li>Plate Control Normalization（板内控制标准化）</li>
</ul>
<script type="math/tex; mode=display">
NPX_{Plate-norm}=NPX-NPX_{Plate-medium}</script><p>调整实验板之间的差异引起的系统性偏差，确保不同版上的样本数据在同一基准。</p>
<ul>
<li>Intensity Normalization（强度标准化）：<script type="math/tex; mode=display">
NPX_{Intensity-norm}=NPX-NPX_{all-medium}</script></li>
</ul>
<p>其中⭐<strong>整体中位数表示的是整体送检样本（不包括外参）的中位数</strong>，目的是消除实验中由于整体信号强度差异引起的变异。</p>
<p>综上，板内控制标准化更适合校正实验板之间的变异而导致的系统性误差；整体强度标准化用于样本之间整体信号强度差异所带来的偏差（<strong>仍需进一步诠释</strong>）。</p>
<p>⭐<strong>选择依据</strong>？</p>
<p>观察NPX数据的分布，当其呈现双峰分布时采用板内控制标准化，以消除板间差异，减少批次效应。</p>
<p>当呈现单峰分布时采用整体强度标准化，此时代表分布较为一致。</p>
<p>（<strong>待完善</strong>）</p>
<h3 id="3-5-OLINK数据"><a href="#3-5-OLINK数据" class="headerlink" title="3.5 OLINK数据"></a>3.5 OLINK数据</h3><h4 id="3-5-1-原始数据构成"><a href="#3-5-1-原始数据构成" class="headerlink" title="3.5.1 原始数据构成"></a>3.5.1 原始数据构成</h4><p>OLINK数据由以下构成，包括<strong>送检样本</strong>以及<strong>外参</strong>。</p>
<p>包括：</p>
<ul>
<li><p>SampleID：样本标识符；</p>
</li>
<li><p>WellID：样本板上标识符；</p>
</li>
<li><p>PlateID：实验板标识符；</p>
</li>
<li><p>OlinkID：蛋白标识符（识别结果）；</p>
</li>
<li><p>Count：目标蛋白检测信号强度（原始数据）；</p>
</li>
<li><p>ExtNPX：NPX值；</p>
</li>
<li><p>AssayQC<strong>（质控）</strong>：蛋白质分析（assay）质量评估指标，评估蛋白质分析过程是否可信。通常包括了信号强度、背景噪音、信号/噪音比、扩展效率、PCR扩增效率。PLINK平台通过计算上述综合得分来评估实验质量；</p>
</li>
<li><p>SampleQC<strong>（质控）</strong>：样本质量控制结果评估，检查样本在Incubation、Extension、Amplification三个阶段是否正常。如：（1）样本Count数<strong>过低</strong>，或样本内参Count数<strong>过低</strong>（FAIL）；（2）样本内参Count数<strong>较</strong>低（WARN）；（3）Plate Control样本中，内参偏离正常范围（FAIL）；（4）Negative Control样本中，较多Assay的Count数高于期望（FAIL）。</p>
<p><img src="/img/loading.gif" data-original="/2024/08/26/proteomics/criteria_olink_sampleqc.png" alt="criteria_olink_sampleqc"></p>
</li>
</ul>
<div class="table-container">
<table>
<thead>
<tr>
<th>SampleID</th>
<th>WellID</th>
<th>PlateID</th>
<th>OlinkID</th>
<th>Count</th>
<th>ExtNPX</th>
<th>AssayQC</th>
<th>SampleQC</th>
</tr>
</thead>
<tbody>
<tr>
<td></td>
<td></td>
<td>plataX-X</td>
<td></td>
<td></td>
<td></td>
<td>PASS/FAIL</td>
<td>PASS/FAIL</td>
</tr>
</tbody>
</table>
</div>
<h4 id="3-5-2-数据组成"><a href="#3-5-2-数据组成" class="headerlink" title="3.5.2 数据组成"></a>3.5.2 数据组成</h4><ul>
<li>样本包括：送检样本+外参，外参=样本控制+板内控制+阴性控制</li>
<li>Assay包括：Unique Protein：互不相同的蛋白+重复测定的蛋白+内参（通常为3*板数），例如有2个蛋白被重复测量3次，则重复测定的数目记作4（2×(3-1)代表已经被测量而又被测量的数目）。</li>
</ul>
<h2 id="4-Nested-Case-control-Study"><a href="#4-Nested-Case-control-Study" class="headerlink" title="4 Nested Case-control Study"></a>4 Nested Case-control Study</h2><p><strong>巢式病例对照研究</strong>：对一个事先确定好的列队进行随访观察的基础上，再应用病例对照研究。其<strong>病例和对照人群都从一个队列研究中选取</strong>。选择对照组时尽量减少混杂因素的影响。</p>
<h1 id="可视化分析"><a href="#可视化分析" class="headerlink" title="可视化分析"></a>可视化分析</h1><h2 id="1-森林图"><a href="#1-森林图" class="headerlink" title="1 森林图"></a>1 森林图</h2><h2 id="2-火山图Volcano"><a href="#2-火山图Volcano" class="headerlink" title="2 火山图Volcano"></a>2 火山图Volcano</h2><p>散点图的一种，将显著性度量（p-value）和变化幅度FC相结合，以便快速直观地识别变化幅度较大且具有统计学意义的数据。</p>
<p><strong>变化幅度FC</strong>：差异倍数，基因在一组样品中的表达值的均值除以其在另一组样品中的表达值的均值，所以火山图只适合两组样品之间的比较。</p>
<p>一般需要做对数转换$Log_2$。</p>
<p>绘制火山图需要包含3列数据。</p>
<div class="table-container">
<table>
<thead>
<tr>
<th>Name</th>
<th>FC</th>
<th>Pvalue</th>
<th>Marker（非必须）</th>
</tr>
</thead>
<tbody>
<tr>
<td>ID1</td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td>ID2</td>
<td></td>
<td></td>
</tr>
</tbody>
</table>
</div>
<h3 id="蛋白质组学中的火山图："><a href="#蛋白质组学中的火山图：" class="headerlink" title="蛋白质组学中的火山图："></a>蛋白质组学中的火山图：</h3><p>蛋白质组学和基因组学分析中的火山图一般是通过比较实验组和对照组的基因或蛋白质表达量，快速识别表达水平上有显著差异的蛋白质或者基因。</p>
<ul>
<li><strong>X轴</strong>：代表对数转换后的倍数变化（$Log_2(FC)$），通常X轴的零点表示基因或蛋白质在实验组和对照组之间没有变化，右侧表示上调表达，左侧表示下调表达；</li>
<li><strong>Y轴</strong>：代表P-value的负对数转换（$-log_{10}(p-value)$），该值越高，代表蛋白质或基因的表达差异越显著。</li>
<li>一般来说，使用不同颜色来标记上调、下调以及不显著的蛋白质。比如，用红色表示显著上调（$log_2(FC)&gt;1$且P-value&lt;0.05），用绿色表示显著上调（$log_2(FC)&lt;-1$且P-value&lt;0.05），用灰色表示不显著变化的蛋白质。</li>
</ul>
<p>火山图中间点数较少（通常代表无显著差异的基因或者蛋白质），两侧的点（显著差异的基因或者蛋白质）。</p>
</article><div class="post-copyright"><div class="post-copyright__author"><span class="post-copyright-meta"><i class="fas fa-circle-user fa-fw"></i>文章作者: </span><span class="post-copyright-info"><a href="http://example.com">远山</a></span></div><div class="post-copyright__type"><span class="post-copyright-meta"><i class="fas fa-square-arrow-up-right fa-fw"></i>文章链接: </span><span class="post-copyright-info"><a href="http://example.com/2024/08/26/proteomics/">http://example.com/2024/08/26/proteomics/</a></span></div><div class="post-copyright__notice"><span class="post-copyright-meta"><i class="fas fa-circle-exclamation fa-fw"></i>版权声明: </span><span class="post-copyright-info">本博客所有文章除特别声明外，均采用 <a href="https://creativecommons.org/licenses/by-nc-sa/4.0/" target="_blank">CC BY-NC-SA 4.0</a> 许可协议。转载请注明来自 <a href="http://example.com" target="_blank">FarMountBiolog</a>！</span></div></div><div class="tag_share"><div class="post-meta__tag-list"><a class="post-meta__tags" href="/tags/StudyNotes/">StudyNotes</a></div><div class="post_share"><div class="social-share" data-image="/img/Kakasi.jpg" data-sites="facebook,twitter,wechat,weibo,qq"></div><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/butterfly-extsrc@1.1.3/sharejs/dist/css/share.min.css" media="print" onload="this.media='all'"><script src="https://cdn.jsdelivr.net/npm/butterfly-extsrc@1.1.3/sharejs/dist/js/social-share.min.js" defer></script></div></div><nav class="pagination-post" id="pagination"><div class="next-post pull-full"><a href="/2024/08/24/Datasets/" title="Bioinformatics Datasets"><div class="cover" style="background: var(--default-bg-color)"></div><div class="pagination-info"><div class="label">下一篇</div><div class="next_info">Bioinformatics Datasets</div></div></a></div></nav><div class="relatedPosts"><div class="headline"><i class="fas fa-thumbs-up fa-fw"></i><span>相关推荐</span></div><div class="relatedPosts-list"><div><a href="/2024/08/24/Datasets/" title="Bioinformatics Datasets"><div class="cover" style="background: var(--default-bg-color)"></div><div class="content is-center"><div class="date"><i class="far fa-calendar-alt fa-fw"></i> 2024-08-24</div><div class="title">Bioinformatics Datasets</div></div></a></div><div><a href="/2024/08/22/Study-Notes-Genomics/" title="Study Notes-Genomics"><div class="cover" style="background: var(--default-bg-color)"></div><div class="content is-center"><div class="date"><i class="far fa-calendar-alt fa-fw"></i> 2024-08-22</div><div class="title">Study Notes-Genomics</div></div></a></div></div></div><hr class="custom-hr"/><div id="post-comment"><div class="comment-head"><div class="comment-headline"><i class="fas fa-comments fa-fw"></i><span> 评论</span></div></div><div class="comment-wrap"><div><div id="gitalk-container"></div></div></div></div></div><div class="aside-content" id="aside-content"><div class="card-widget card-info"><div class="is-center"><div class="avatar-img"><img src="/img/loading.gif" data-original="/img/Kakasi.jpg" onerror="this.onerror=null;this.src='/img/friend_404.gif'" alt="avatar"/></div><div class="author-info__name">远山</div><div class="author-info__description">大鹏一日同风起，扶摇直上九万里</div></div><div class="card-info-data site-data is-center"><a href="/archives/"><div class="headline">文章</div><div class="length-num">4</div></a><a href="/tags/"><div class="headline">标签</div><div class="length-num">2</div></a><a href="/categories/"><div class="headline">分类</div><div class="length-num">0</div></a></div><a id="card-info-btn" target="_blank" rel="noopener" href="https://github.com/BenjaminMount"><i class="fab fa-github"></i><span>Follow Me</span></a><div class="card-info-social-icons is-center"><a class="social-icon" href="https://github.com/BenjaminMount/" target="_blank" title="Github"><i class="fab fa-github"></i></a><a class="social-icon" href="tencent://AddContact/?fromId=45&amp;fromSubId=1&amp;subcmd=all&amp;uin=2396979300&amp;website=www.oicqzone.com" target="_blank" title="QQ"><i class="fab fa-qq"></i></a><a class="social-icon" href="mailto:liutongyan@hust.edu.cn" target="_blank" title="Email"><i class="fas fa-envelope-open-text"></i></a></div></div><div class="card-widget card-announcement"><div class="item-headline"><i class="fas fa-bullhorn fa-shake"></i><span>公告</span></div><div class="announcement_content">Keep thinking. Progress itself is hard.</div></div><div class="sticky_layout"><div class="card-widget" id="card-toc"><div class="item-headline"><i class="fas fa-stream"></i><span>目录</span><span class="toc-percentage"></span></div><div class="toc-content is-expand"><ol class="toc"><li class="toc-item toc-level-1"><a class="toc-link" href="#Proteomics"><span class="toc-text">Proteomics</span></a><ol class="toc-child"><li class="toc-item toc-level-2"><a class="toc-link" href="#0-Preface"><span class="toc-text">0 Preface</span></a><ol class="toc-child"><li class="toc-item toc-level-3"><a class="toc-link" href="#%E6%95%B0%E6%8D%AE%E8%8E%B7%E5%8F%96%E4%B8%8E%E9%A2%84%E5%A4%84%E7%90%86"><span class="toc-text">数据获取与预处理</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#%E6%95%B0%E6%8D%AE%E5%AE%9A%E9%87%8F%E5%88%86%E6%9E%90"><span class="toc-text">数据定量分析</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#%E6%95%B0%E6%8D%AE%E6%A0%87%E5%87%86%E5%8C%96%E5%92%8C%E5%BD%92%E4%B8%80%E5%8C%96"><span class="toc-text">数据标准化和归一化</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#%E5%B7%AE%E5%BC%82%E8%A1%A8%E8%BE%BE%E5%88%86%E6%9E%90"><span class="toc-text">差异表达分析</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#%E5%8A%9F%E8%83%BD%E6%B3%A8%E9%87%8A%E4%B8%8E%E9%80%9A%E8%B7%AF%E5%88%86%E6%9E%90"><span class="toc-text">功能注释与通路分析</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#%E7%BB%93%E6%9E%9C%E9%AA%8C%E8%AF%81%E4%B8%8E%E8%A7%A3%E9%87%8A"><span class="toc-text">结果验证与解释</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#%E6%95%B0%E6%8D%AE%E5%8F%AF%E8%A7%86%E5%8C%96%E4%B8%8E%E6%8A%A5%E5%91%8A"><span class="toc-text">数据可视化与报告</span></a></li></ol></li><li class="toc-item toc-level-2"><a class="toc-link" href="#1-%E6%89%B9%E6%AC%A1%E6%95%88%E5%BA%94"><span class="toc-text">1 批次效应</span></a><ol class="toc-child"><li class="toc-item toc-level-3"><a class="toc-link" href="#%E5%BD%B1%E5%93%8D%EF%BC%9A"><span class="toc-text">影响：</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#%E6%A0%A1%E6%AD%A3%E6%96%B9%E6%B3%95%EF%BC%9A"><span class="toc-text">校正方法：</span></a></li></ol></li><li class="toc-item toc-level-2"><a class="toc-link" href="#2-%E5%A4%9A%E9%87%8D%E6%A3%80%E9%AA%8C%E6%A0%A1%E6%AD%A3"><span class="toc-text">2 多重检验校正</span></a><ol class="toc-child"><li class="toc-item toc-level-3"><a class="toc-link" href="#2-1-Bonferroni"><span class="toc-text">2.1 Bonferroni</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#2-2-FalseDiscovery-Rate"><span class="toc-text">2.2 FalseDiscovery Rate</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#2-3-ENT"><span class="toc-text">2.3 ENT</span></a></li></ol></li><li class="toc-item toc-level-2"><a class="toc-link" href="#3-%E2%AD%90%E2%AD%90OLINK%E6%8A%80%E6%9C%AF"><span class="toc-text">3 ⭐⭐OLINK技术</span></a><ol class="toc-child"><li class="toc-item toc-level-3"><a class="toc-link" href="#3-1-OLINK"><span class="toc-text">3.1 OLINK</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#3-2-OLINK-Explore-HT"><span class="toc-text">3.2 OLINK Explore HT</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#3-3-%E6%B5%81%E7%A8%8B%E8%AE%BE%E7%BD%AE%E4%BB%A5%E5%8F%8A%E6%93%8D%E4%BD%9C%E7%BB%86%E8%8A%82"><span class="toc-text">3.3 流程设置以及操作细节</span></a><ol class="toc-child"><li class="toc-item toc-level-4"><a class="toc-link" href="#3-3-1-Plate"><span class="toc-text">3.3.1 Plate</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#3-3-2-%E2%AD%90%E5%AD%94%E4%B8%8A%E5%A4%96%E5%8F%82"><span class="toc-text">3.3.2 ⭐孔上外参</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#3-3-3-Block%E7%A8%80%E9%87%8A"><span class="toc-text">3.3.3 Block稀释</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#3-3-4-%E2%AD%90%E6%A0%B7%E6%9C%AC%E5%86%85%E5%8F%82"><span class="toc-text">3.3.4 ⭐样本内参</span></a></li></ol></li><li class="toc-item toc-level-3"><a class="toc-link" href="#3-4-OLINK-NPX%E6%95%B0%E6%8D%AE"><span class="toc-text">3.4 OLINK NPX数据</span></a><ol class="toc-child"><li class="toc-item toc-level-4"><a class="toc-link" href="#3-4-1-NPX%E6%95%B0%E6%8D%AE%E6%A6%82%E5%BF%B5"><span class="toc-text">3.4.1 NPX数据概念</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#3-4-2-NPX%E6%95%B0%E6%8D%AE%E6%A0%87%E5%87%86%E5%8C%96"><span class="toc-text">3.4.2 NPX数据标准化</span></a></li></ol></li><li class="toc-item toc-level-3"><a class="toc-link" href="#3-5-OLINK%E6%95%B0%E6%8D%AE"><span class="toc-text">3.5 OLINK数据</span></a><ol class="toc-child"><li class="toc-item toc-level-4"><a class="toc-link" href="#3-5-1-%E5%8E%9F%E5%A7%8B%E6%95%B0%E6%8D%AE%E6%9E%84%E6%88%90"><span class="toc-text">3.5.1 原始数据构成</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#3-5-2-%E6%95%B0%E6%8D%AE%E7%BB%84%E6%88%90"><span class="toc-text">3.5.2 数据组成</span></a></li></ol></li></ol></li><li class="toc-item toc-level-2"><a class="toc-link" href="#4-Nested-Case-control-Study"><span class="toc-text">4 Nested Case-control Study</span></a></li></ol></li><li class="toc-item toc-level-1"><a class="toc-link" href="#%E5%8F%AF%E8%A7%86%E5%8C%96%E5%88%86%E6%9E%90"><span class="toc-text">可视化分析</span></a><ol class="toc-child"><li class="toc-item toc-level-2"><a class="toc-link" href="#1-%E6%A3%AE%E6%9E%97%E5%9B%BE"><span class="toc-text">1 森林图</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#2-%E7%81%AB%E5%B1%B1%E5%9B%BEVolcano"><span class="toc-text">2 火山图Volcano</span></a><ol class="toc-child"><li class="toc-item toc-level-3"><a class="toc-link" href="#%E8%9B%8B%E7%99%BD%E8%B4%A8%E7%BB%84%E5%AD%A6%E4%B8%AD%E7%9A%84%E7%81%AB%E5%B1%B1%E5%9B%BE%EF%BC%9A"><span class="toc-text">蛋白质组学中的火山图：</span></a></li></ol></li></ol></li></ol></div></div><div class="card-widget card-recent-post"><div class="item-headline"><i class="fas fa-history"></i><span>最新文章</span></div><div class="aside-list"><div class="aside-list-item no-cover"><div class="content"><a class="title" href="/2024/08/26/proteomics/" title="Proteomics">Proteomics</a><time datetime="2024-08-26T03:48:19.000Z" title="发表于 2024-08-26 11:48:19">2024-08-26</time></div></div><div class="aside-list-item no-cover"><div class="content"><a class="title" href="/2024/08/24/Datasets/" title="Bioinformatics Datasets">Bioinformatics Datasets</a><time datetime="2024-08-24T08:54:54.000Z" title="发表于 2024-08-24 16:54:54">2024-08-24</time></div></div><div class="aside-list-item no-cover"><div class="content"><a class="title" href="/2024/08/22/Reading-Notes-Mathematical-Population-Genetics-by-Warren-J-Ewens/" title="Reading Notes-&lt;Mathematical Population Genetics&gt; by Warren J.Ewens">Reading Notes-&lt;Mathematical Population Genetics&gt; by Warren J.Ewens</a><time datetime="2024-08-22T06:13:03.000Z" title="发表于 2024-08-22 14:13:03">2024-08-22</time></div></div><div class="aside-list-item no-cover"><div class="content"><a class="title" href="/2024/08/22/Study-Notes-Genomics/" title="Study Notes-Genomics">Study Notes-Genomics</a><time datetime="2024-08-22T06:12:32.000Z" title="发表于 2024-08-22 14:12:32">2024-08-22</time></div></div></div></div></div></div></main><footer id="footer" style="background-image: url('/img/pic_1.png')"><div id="footer-wrap"><div class="copyright">&copy;2024 By 远山</div><div class="framework-info"><span>框架 </span><a target="_blank" rel="noopener" href="https://hexo.io">Hexo</a><span class="footer-separator">|</span><span>主题 </span><a target="_blank" rel="noopener" href="https://github.com/jerryc127/hexo-theme-butterfly">Butterfly</a></div></div></footer></div><div id="rightside"><div id="rightside-config-hide"><button id="readmode" type="button" title="阅读模式"><i class="fas fa-book-open"></i></button><button id="darkmode" type="button" title="浅色和深色模式转换"><i class="fas fa-adjust"></i></button><button id="hide-aside-btn" type="button" title="单栏和双栏切换"><i class="fas fa-arrows-alt-h"></i></button></div><div id="rightside-config-show"><button id="rightside-config" type="button" title="设置"><i class="fas fa-cog fa-spin"></i></button><button class="close" id="mobile-toc-button" type="button" title="目录"><i class="fas fa-list-ul"></i></button><a id="to_comment" href="#post-comment" title="直达评论"><i class="fas fa-comments"></i></a><button id="go-up" type="button" title="回到顶部"><span class="scroll-percent"></span><i class="fas fa-arrow-up"></i></button></div></div><div><script src="/js/utils.js?v=4.13.0"></script><script src="/js/main.js?v=4.13.0"></script><script src="https://cdn.jsdelivr.net/npm/medium-zoom@1.1.0/dist/medium-zoom.min.js"></script><div class="js-pjax"><script>if (!window.MathJax) {
  window.MathJax = {
    tex: {
      inlineMath: [['$', '$'], ['\\(', '\\)']],
      tags: 'ams'
    },
    chtml: {
      scale: 1.1
    },
    options: {
      renderActions: {
        findScript: [10, doc => {
          for (const node of document.querySelectorAll('script[type^="math/tex"]')) {
            const display = !!node.type.match(/; *mode=display/)
            const math = new doc.options.MathItem(node.textContent, doc.inputJax[0], display)
            const text = document.createTextNode('')
            node.parentNode.replaceChild(text, node)
            math.start = {node: text, delim: '', n: 0}
            math.end = {node: text, delim: '', n: 0}
            doc.math.push(math)
          }
        }, '']
      }
    }
  }
  
  const script = document.createElement('script')
  script.src = 'https://cdn.jsdelivr.net/npm/mathjax@3.2.2/es5/tex-mml-chtml.min.js'
  script.id = 'MathJax-script'
  script.async = true
  document.head.appendChild(script)
} else {
  MathJax.startup.document.state(0)
  MathJax.texReset()
  MathJax.typesetPromise()
}</script><script>(() => {
  const initGitalk = () => {
    const gitalk = new Gitalk(Object.assign({
      clientID: 'Ov23liSwPtc8jModvEnf',
      clientSecret: '087679c1568bceccc832c27ef3c219a01cb93a8e',
      repo: 'BenjaminMount.github.io',
      owner: 'BenjaminMount',
      admin: ['BenjaminMount'],
      id: '6ff3dec17148e4e0f12981264184c754',
      updateCountCallback: commentCount
    },null))

    gitalk.render('gitalk-container')
  }

  const loadGitalk = async() => {
    if (typeof Gitalk === 'function') initGitalk()
    else {
      await getCSS('https://cdn.jsdelivr.net/npm/gitalk@1.8.0/dist/gitalk.min.css')
      await getScript('https://cdn.jsdelivr.net/npm/gitalk@1.8.0/dist/gitalk.min.js')
      initGitalk()
    }
  }
  
  const commentCount = n => {
    const isCommentCount = document.querySelector('#post-meta .gitalk-comment-count')
    if (isCommentCount) {
      isCommentCount.textContent= n
    }
  }

  if ('Gitalk' === 'Gitalk' || !true) {
    if (true) btf.loadComment(document.getElementById('gitalk-container'), loadGitalk)
    else loadGitalk()
  } else {
    window.loadOtherComment = loadGitalk
  }
})()</script></div><script id="canvas_nest" defer="defer" color="0,0,255" opacity="0.7" zIndex="-1" count="99" mobile="false" src="https://cdn.jsdelivr.net/npm/butterfly-extsrc@1.1.3/dist/canvas-nest.min.js"></script><script src="https://cdn.jsdelivr.net/npm/butterfly-extsrc@1.1.3/dist/activate-power-mode.min.js"></script><script>POWERMODE.colorful = true;
POWERMODE.shake = false;
POWERMODE.mobile = false;
document.body.addEventListener('input', POWERMODE);
</script><script async data-pjax src="//busuanzi.ibruce.info/busuanzi/2.3/busuanzi.pure.mini.js"></script><div id="local-search"><div class="search-dialog"><nav class="search-nav"><span class="search-dialog-title">搜索</span><span id="loading-status"></span><button class="search-close-button"><i class="fas fa-times"></i></button></nav><div class="is-center" id="loading-database"><i class="fas fa-spinner fa-pulse"></i><span>  数据库加载中</span></div><div class="search-wrap"><div id="local-search-input"><div class="local-search-box"><input class="local-search-box--input" placeholder="搜索文章" type="text"/></div></div><hr/><div id="local-search-results"></div><div id="local-search-stats-wrap"></div></div></div><div id="search-mask"></div><script src="/js/search/local-search.js?v=4.13.0"></script></div></div>
        <style>
            [bg-lazy] {
                background-image: none !important;
                background-color: #eee !important;
            }
        </style>
        <script>
            window.imageLazyLoadSetting = {
                isSPA: false,
                preloadRatio: 1,
                processImages: null,
            };
        </script><script>window.addEventListener("load",function(){var t=/\.(gif|jpg|jpeg|tiff|png)$/i,r=/^data:image\/[a-z]+;base64,/;Array.prototype.slice.call(document.querySelectorAll("img[data-original]")).forEach(function(a){var e=a.parentNode;"A"===e.tagName&&(e.href.match(t)||e.href.match(r))&&(e.href=a.dataset.original)})});</script><script>(r=>{r.imageLazyLoadSetting.processImages=t;var a=r.imageLazyLoadSetting.isSPA,o=r.imageLazyLoadSetting.preloadRatio||1,d=i();function i(){var t=Array.prototype.slice.call(document.querySelectorAll("img[data-original]")),e=Array.prototype.slice.call(document.querySelectorAll("[bg-lazy]"));return t.concat(e)}function t(t){(a||t)&&(d=i());for(var e,n=0;n<d.length;n++)0<=(e=(e=d[n]).getBoundingClientRect()).bottom&&0<=e.left&&e.top<=(r.innerHeight*o||document.documentElement.clientHeight*o)&&(()=>{var t,e,a,o,i=d[n];e=function(){d=d.filter(function(t){return i!==t}),r.imageLazyLoadSetting.onImageLoaded&&r.imageLazyLoadSetting.onImageLoaded(i)},(t=i).dataset.loaded||(t.hasAttribute("bg-lazy")?(t.removeAttribute("bg-lazy"),e&&e()):(a=new Image,o=t.getAttribute("data-original"),a.onload=function(){t.src=o,t.removeAttribute("data-original"),t.setAttribute("data-loaded",!0),e&&e()},a.onerror=function(){t.removeAttribute("data-original"),t.setAttribute("data-loaded",!1),t.src=o},t.src!==o&&(a.src=o)))})()}function e(){clearTimeout(t.tId),t.tId=setTimeout(t,500)}t(),document.addEventListener("scroll",e),r.addEventListener("resize",e),r.addEventListener("orientationchange",e)})(this);</script></body></html>